### ISSUE TITLE
Cannot find MDK Settings 

### ISSUE THREAD

**Issue by malware-dev** [malware-dev/MDK-SE#70]
Some people are reporting issues where MDK is saying that it cannot find the MDK settings because they're corrupted. It has usually been an outside problem, something that isn't caused by MDK itself. But it's starting to crop up often enough that it's becoming an issue, so I'd like to make a compilation of solutions and troubleshooting for this issue here. If there _is_ something I can do with MDK to alleviate this problem, I would like to do so.

Please enable and find the activity log after a crash, by following these instructions.
https://blogs.msdn.microsoft.com/visualstudio/2010/02/24/troubleshooting-extensions-with-the-activity-log/

**Comment by malware-dev** [comment]
https://github.com/malware-dev/MDK-SE/issues/69

https://github.com/malware-dev/MDK-SE/issues/66

**Comment by KreyserWaryag** [comment]
[pack.zip](https://github.com/malware-dev/MDK-SE/files/2215076/pack.zip)

**Comment by malware-dev** [comment]
@KreyserWaryag Thank you.

Great. A _third_ cause.

**Comment by KreyserWaryag** [comment]
Thanks. Here is the log:
[pack.zip](https://github.com/malware-dev/MDK-SE/files/2216888/pack.zip)

**Comment by malware-dev** [comment]
@KreyserWaryag That logs show no error any more. Does this mean that it now works correctly for you?

**Comment by KreyserWaryag** [comment]
No. It's not working. However, the error is already different (I updated VS)
![image](https://user-images.githubusercontent.com/41488272/43045699-07b56db0-8dc6-11e8-8085-23c756ab61a6.png)
Log: 
[pack.zip](https://github.com/malware-dev/MDK-SE/files/2217021/pack.zip)

**Comment by KreyserWaryag** [comment]
[pack.zip](https://github.com/malware-dev/MDK-SE/files/2217033/pack.zip)
this log is made using Your [extension](https://github.com/malware-dev/MDK-SE/releases/tag/1.1.13)

**Comment by malware-dev** [comment]
I'm sorry. I just don't understand this. Core assemblies belonging to Visual Studio is failing for you and I cannot understand why.

**Comment by KreyserWaryag** [comment]
Sorry to bother you. This is really the problem VS. I just tried to create a console application-the error is the same (. Sorry again. I will try to reinstall VS.

**Comment by malware-dev** [comment]
Yes, that usually proves to be the problem with this error. I was just hoping to find a solution.

**Comment by KreyserWaryag** [comment]
HURRAH. [This ](https://developercommunity.visualstudio.com/content/problem/94979/cannot-open-any-projects-in-solution-after-install.html)helped me.

**Comment by malware-dev** [comment]
Thank you for letting me know!

**Comment by WhiteHusky** [comment]
I encountered the same issue, but I managed to resolve it by updating my copy of Visual Studio 2017.
So that might be a solution for some, but this was a fresh install of MDK-SE.

**Comment by malware-dev** [comment]
Yes, MDK requires Visual Studio 2017 15.6 or better. The officially documented way to set this requirement, however... doesn't work.

**Comment by malware-dev** [comment]
@craftersmine Good question... 🙁

**Comment by craftersmine** [comment]
@malware-dev Is it possible to somehow add programmno in the code an exact check of the version of Visual Studio? For example, if the addin is installed, then inside the plugin check that this version is suitable and display a message if the version does not fit? As I understood in the manifest of the plug-in to specify a more accurate version than 15.0 it is impossible

**Comment by malware-dev** [comment]
@craftersmine I've tried. But I can't get the correct version number. I just get that 15.0.

**Comment by jturp** [comment]
For me, this happens when swapping between two PCs using the same scripts. I tell PC 1 to "Upgrade MDK to make everything work properly", and then forget to redo PC 2. If I then try to deploy a script from PC 2, it gives me the "Unable to find MDK files" - because they are now pointing to a location on a different PC. If I close / reopen VS on PC 2, and upgrade again, everything works fine.

**Comment by malware-dev** [comment]
@viveleltsi Your Visual Studio is too old, as noted on the[ Releases page here](https://github.com/malware-dev/MDK-SE/releases) you need visual studio 15.8.

**Comment by pedrom505** [comment]
Dear All,

I'm having the same problem to use the MDK extension.
Has this problem a solution, or the solution was not found yet?
I'm a brazilian developer!

**Comment by malware-dev** [comment]
Same answer to you as well, @pedrom505. Update your Visual Studio.

**Comment by pedrom505** [comment]
But i'm using the last version of visual studio 2017. In my machine is installed the visual studio 2017 (version) 15.9.7.
I think that the last version of visual studio doesn't solve the problem.
Do you know where is the MDK settings file? My visual studio isn't find this file.

**Comment by malware-dev** [comment]
I'm afraid I can't help you then. I have no idea why it would not work, and you're the first in several months who's had this problem.

**Comment by malware-dev** [comment]
Please upload the activity log as described on top of this page, @pedrom505 .

**Comment by space-bits** [comment]
Hi, @malware-dev I was just installing VS 2017 (v15.9.10) and the MDK extension, release 1.1.30 and noticed that the tag 1.1.30 - released on March 1, named "Major Release Compatibility Patch" - points to cd51a07121baab59888db22dee0bf3581cc75986 which seems to be a merge from last October, but the commit with message "Major Release Compatibility Patch" from March 1 points to f9c23120bebf3cd0876ba483945f6d503dd35d0a

Forgive me if I'm misunderstanding what's going on here, but I thought that this might be accidental and somehow the source of the "Cannot find MDK Settings" issue, which I am also having after a fresh install of both VS and the MDK extension.

Here's my activity log for what it's worth 😄 
[ActivityLog.zip](https://github.com/malware-dev/MDK-SE/files/3025313/ActivityLog.zip)

**Comment by malware-dev** [comment]
I'm sorry to disappoint, @space-bits, but there's been over 800 downloads of version 1.1.30, and this particular issue seems to happen to only a few people. I have _no_ idea why and the log doesn't really help in any way. I'm not even finding any reference to MDK at all in there. I have no clue why some people get this problem. I'm now running the latest VS myself.

I agree the tag seems a little strange, I haven't seen that myself and I couldn't tell you why it does that - but it's not really relevant. The VSIX file is manually built with the sources as I have them on my machine. What that points to is unimportant to that.

**Comment by space-bits** [comment]
That's unfortunate, but I'll keep playing around with it.

What's odd is that in VS MDK is shown in the tools dropdown, and it the blueprint manager + check for updates work fine. It's only unable to find the location of the settings when creating a new project.

Either way, the effort that you've put into this is much appreciated, and if I find a solution, I'll post back here so that maybe others can follow.

**Comment by malware-dev** [comment]
@space-bits If you can actually figure out what's going on and let me know so I can finally _fix it_ that would be fantastic. Thank you, and I _really_ hope you find out. It's practically impossible to find the cause of an issue one does not experience oneself and I'm really sorry you're having problems.

**Comment by craftersmine** [comment]
Ooohhh... Why Microsoft couldn't make easy way to get full version of Visual Studio? It will help many others extension developers

**Comment by space-bits** [comment]
@malware-dev If I do come across a solution, I'll let you know what I did for sure, I'm normally developing on Linux so VS is a tad new to me anyway. Fingers crossed though. Again, thanks for the effort you've put into the programming guide as well as the extension.

**Comment by malware-dev** [comment]
@craftersmine I'm just using Community Edition to write MDK. It works quite nicely.

**Comment by craftersmine** [comment]
@malware-dev, no, I mean the numeric version, like 15.7.8, etc

**Comment by malware-dev** [comment]
@craftersmine Oh, yeah, that. Absolutely.

**Comment by MrChoke72** [comment]
I have this error too.  Fresh install of VS 2017.

**Comment by LukeChrisStark** [comment]
Hi! I've tried using MDK, but every time I try to create a new ingame script project I get the "Cannot find MDK settings" error. I've tried using both Visual Studio Community 2017 and 2019, both updated to their latest versions, neither worked. I tried the gacutil removal of the Microsoft.Build.Framework, both with and without re-adding it afterwards, neither way worked.

Here's my activity log:  
[ActivityLog.zip](https://github.com/malware-dev/MDK-SE/files/4485463/ActivityLog.zip)

**Comment by malware-dev** [comment]
@LukeChrisStark I'm afraid there's nothing in the log that is helpful. I still have no clue why this happens to some people.

Where are you creating your projects? Could there be unusual symbols in the path? Or is the final path excessively long?

**Comment by LukeChrisStark** [comment]
The path I'm trying to use is "C:\Users\5842\source\repos", so no unusual symbols. Visual Studio creates a folder "C:\Users\5842\source\repos\IngameScript1\IngameScript1", but it is empty of any files. My Game Install Path & Script Output Path also have no unusual characters in them.

I'll try installing it on another machine and I'll get back to you.

Also, I'm just guessing here, but could it be a problem that the extension is called "MDK **/** SE"? Could the slash in the extension name cause problems?

**Comment by malware-dev** [comment]
@LukeChrisStark Then this would be a much more common problem... but it only affects a very few people.

**Comment by LukeChrisStark** [comment]
Oh, ok. Thanks for the information!

I've installed MDK on another machine and it works there. No idea why, though, as pretty much everything is the same on both computers.

Thanks for your time!

**Comment by malware-dev** [comment]
@LukeChrisStark If you actually manage to find out what the heck is going on, I'd appreciate if you'd let me know.

**Comment by Mutzi100** [comment]
I found one cause for such errors.
@LukeChrisStark Do/did you have Visual Studio 2008 installed not your PC?
As describe here: https://developercommunity.visualstudio.com/content/problem/430365/deploying-an-azure-resource-group-deployproj-fails-1.html, VS 2008 adds outdatet assemblies to the gac.

This is not MDKs fault. What would help debugging, is: On line 151 in the IngameScriptWizard.cs output the message of the COMException.

**Comment by LukeChrisStark** [comment]
@malware-dev Sure, if I can figure it out I'll post here.

@Mutzi100 No, I never had VS2008 on either machine, so that's not the cause for me.

**Comment by craftersmine** [comment]
> @Mutzi100 outdatet assemblies to the gac.

Maybe the cause of this error is that destination machine has older versions of .NET Framework? For example 4.7 installed firstly, then user needed 3.5 version and installed it, which downgraded some assemblies? But that just a thought...

**Comment by Lazrius** [comment]
I was having the same issue and it has plagued me for the better part of a day. I eventually got it working by doing a lot of different things, but I don't know which ones actually solved my issue.

- I deleted all other versions of Visual Studio I had (I had 2005, 2008, 2015, 2017, 2019).
- Updated 2019 to the latest version.
- Ran a repair on the install to be safe and refresh.
- Used gacutil.exe to uninstall Microsoft.Build.Framework **and** Microsoft.VisualStudio.Shell.Interop.dll.
- Completely purge using settings and all extensions, then reinstall via the VSIX.

Don't know if this will help anyone, but it solved my problem.
